src/query/service/src/sql/planner/binder/bind_context.rs

```diff
@@ -34,6 +34,12 @@ use crate::sql::optimizer::SExpr;
 use crate::sql::plans::Scalar;
 use crate::sql::NameResolutionContext;
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum InVisibility {
+    UsingColumnOfJoin,
+    InnerColumnOfStruct,
+}
```

Comment:
Currently, there are two cases where the column needs to be invisible:

First, `using(a)` needs to ensure that a table's `a` is not visible in `select *`.

Second, the inner columns of struct need to ensure that all columns are invisible in `select *`.

There is another particular case, `using(alias_name)`, such as the following SQL:
```
select a from (select number as a from numbers(3)) n join (select number as a from numbers(4)) nn using (a) order by a;
```

`a` is an alias for `number`, there is no qualified table name specified for `a`, so we need to filter an `a` in `match_column_binding` to avoid column reference being ambiguous.

We don't need to set visibility for the `QualifidWildcard`, because it has a specific table name and will naturally be handled correctly in `match_column_binding`.

Reply:
@xudong963 We actually haven't supported qualified wildcard yet, it's necessary to mark the `USING` column as visible.

Reply:
> `a` is an alias for `number`, there is no qualified table name specified for `a`, so we need to filter an `a` in `match_column_binding` to avoid column reference being ambiguous.

The table name of `number as a` should be `nn`?

Reply:
> @xudong963 We actually haven't supported qualified wildcard yet, it's necessary to mark the `USING` column as visible.

Oh, I meant such format: `select t.a`.

Reply:
> The table name of `number as a` should be `nn`?

Yes, in `ColumnBinding` the table name is `nn`.  But in `Expr`(ast) the table name is `None`.

So in `match_column_binding`,  so we need to filter an `a` to avoid column reference being ambiguous.

Reply:
> So in `match_column_binding`, so we need to filter an `a` to avoid column reference being ambiguous.

We can imply that if a column is **visible in qualified wildcard**, we will choose the first candidate for an unqualified reference. Both `USING` and `NATURAL JOIN` follow this behavior, it's better to use a more common name to represent the visibility.